faunadb-scala/src/main/scala/faunadb/values/time/package.scala

```diff
@@ -0,0 +1,89 @@
+package faunadb.values
+
+import org.joda.time.Instant
+import org.joda.time.format.ISODateTimeFormat
+import scala.Ordered._
+
+package time {
+
+  /**
+    * Represents an instance of [[org.joda.time.Instant]] with nanosecond precision.
+    * It calculates the overflow from nano to milliseconds and adds to the initial instant.
+    */
+  class HighPrecisionTime private(initialInstant: Instant, nanosToAdd: Long) {
+
+    /** [[org.joda.time.Instant]] representation. Truncates micro and nanoseconds. */
+    val instant = initialInstant.plus(nanosToAdd / 1000000)
+
+    /** Nanoseconds added to the initial time */
+    val nanos = nanosToAdd % 1000000
+
+    /** Microseconds added to the initial time. Truncates nanoseconds. */
+    val micros = nanos / 1000
+
+    /** Milliseconds since Java epoch. Truncates micro and nanoseconds. */
+    val millis = instant.getMillis
+
+    override def equals(obj: Any): Boolean = obj match {
+      case other: HighPrecisionTime => instant == other.instant && nanos == other.nanos
+      case _                        => false
+    }
+
+    override def hashCode(): Int =
+      instant.hashCode() + nanos.hashCode()
+
+    override def toString =
+      "%s%06dZ".format(
+        instant.toString(ISODateTimeFormat.dateHourMinuteSecondMillis()),
+        nanos
+      )
+  }
+
+  object HighPrecisionTime {
+
+    /**
+      * Creates an instance of [[faunadb.values.time.HighPrecisionTime]] adding
+      * the the micro and nanoseconds informed.
+      */
+    def apply(initial: Instant, microsToAdd: Long = 0, nanosToAdd: Long = 0): HighPrecisionTime =
```

Comment:
This constructor comes handy as we parse the timestamp from the server: `case Precision(micros, nanos) => HighPrecisionTime(toInstant(value), micros.toInt, nanos.toInt)`

Would you prefer different methods rather than `apply` for this? I'm not sure if I got it.